### PR TITLE
Handle write operation using arrow-flight.

### PR DESCRIPTION
**Pull request by Mohammad-nassar10**
Signed-off-by: mohammad-nassar10 <mohammad.nassar@ibm.com>

### CODE REVIEW

File: abm/connector.py [hunk 1]
```diff
@@ -313,3 +313,21 @@ def write_dataset(self, fptr, length):
         tmp_catalog.close()
         # TODO: Need to figure out how to handle error return
         return True
+
+    def write_dataset_bytes(self, bytes):
```
Comment (cdoron) [inline abm/connector.py:317]: Hi. I see that write_dataset_bytes() is similar to write_dataset().
Let's discuss the difference.
Is there a way for one of these methods to call the other, to reduce the code replication?

File: abm/server.py [hunk 2]
```diff
@@ -145,6 +145,24 @@ def do_get(self, context, ticket: fl.Ticket):
...
+        asset_name = json.loads(descriptor.command)['asset']
+        logger.info('getting flight information',
+            extra={'command': descriptor.command,
+                   DataSetID: asset_name,
+                   ForUser: True})
+        with Config(self.config_path) as config:
+            df_bytes = []
+            asset_conf = config.for_asset(asset_name)
+            connector = GenericConnector(asset_conf, logger, self.workdir)
+            batches = reader.read_all().combine_chunks().to_batches(max_chunksize=1)
```
Comment (cdoron) [inline abm/server.py:161]: Please explain this line to me.

**Review by cdoron — COMMENTED** [review]
Looks good.
Let's do a code review. Thanks

File: abm/server.py (hunk 2 above)
Comment (Mohammad-nassar10) [reply abm/server.py:161]: This line gets a list of `RecordBatche` objects from a Table, where each `Recordbatch` represents one row.

File: abm/connector.py (hunk 1 above)
Comment (Mohammad-nassar10) [reply abm/connector.py:317]: Each function gets a different parameters, but maybe we can reduce the code duplication.

**Review by revit13 — APPROVED** [review]

File: abm/server.py [hunk 3]
```diff
@@ -50,9 +51,13 @@ def do_GET(self):
                 self.send_response(HTTPStatus.BAD_REQUEST)
                 self.end_headers()
 
-# Have the same routine for PUT and POST
+    # Have the same routine for PUT and POST
     def do_WRITE(self):
-        logger.info('write requested')
+        self.data_string = self.rfile.read(int(self.headers['Content-Length']))
```
[the remark this answers is not in the document]
Comment (revit13) [reply abm/server.py]: @cdoron we fixed that. Thanks

File: abm/server.py [hunk 4]
```diff
@@ -145,6 +153,28 @@ def do_get(self, context, ticket: fl.Ticket):
...
+        asset_name = json.loads(descriptor.command)['asset']
+        schema = json.loads(descriptor.command)['schema']
+        logger.info('getting flight information',
+            extra={'command': descriptor.command,
+                   DataSetID: asset_name,
+                   ForUser: True})
+        with Config(self.config_path) as config:
+            asset_conf = config.for_asset(asset_name)
+            connector = GenericConnector(asset_conf, logger, self.workdir)
+            batches = reader.read_all().combine_chunks().to_batches(max_chunksize=1)
```
[the remark this answers is not in the document]
Comment (revit13) [reply abm/server.py]: @cdoron we fixed that. Thanks

File: abm/connector.py [hunk 5]
```diff
@@ -267,49 +267,54 @@ def get_dataset_batches(self, schema):
...
                   ' --catalog ' + self.name_in_container(tmp_catalog.name)
         s, container = self.open_socket_to_container(command)
 
-        # eg echo payload | docker run -v /Users/eliot/temp:/local -i airbyte/destination-local-json write --catalog /local/airbyte_catalog.txt --config /local/airbyte_write1.json
         bytesToWrite = length
         while bytesToWrite > 0:
             readSize = CHUNKSIZE if (bytesToWrite - CHUNKSIZE) >= 0 else bytesToWrite
             bytesToWrite -= readSize
             payload = fptr.read(int(readSize))
+            print(payload)
```
Comment (cdoron) [inline abm/connector.py:305]: Where is this printed?
I think this line should be removed

File: abm/server.py [hunk 6]
```diff
@@ -145,6 +146,38 @@ def do_get(self, context, ticket: fl.Ticket):
         # return dataset as arrow flight record batches
         return fl.GeneratorStream(schema, batches)
 
+    '''
+    Serve arrow flight do_put requests
+    '''
+    def do_put(self, context, descriptor, reader, writer):
+        asset_name = json.loads(descriptor.command)['asset']
+        schema = json.loads(descriptor.command)['schema']
```
Comment (cdoron) [inline abm/server.py:154]: No need to run `json.loads(descriptor.command)` twice.
Please check whether the 'asset' and 'schema' fields appear in the map, and return an error if needed

File: fybrik/README.md [hunk 7]
```diff
@@ -53,3 +58,52 @@ You will need a copy of the Fybrik repository (`git clone https://github.com/fyb
...
+When you're finished experimenting with a sample, you may clean up as follows:
+
+Delete the namespace created for this sample:
+
+```bash
+kubectl delete namespace fybrik-airbyte-sample
+```
+
+To experiment with a sample after the deletion of `fybrik-airbyte-sample` namespace,
+re-create the namspace with the following commands and continue from step 6 in the chosen sample.
```
Comment (cdoron) [inline fybrik/README.md:102]: namspace -> namespace

File: fybrik/README.md [hunk 8]
```diff
@@ -33,13 +33,18 @@ You will need a copy of the Fybrik repository (`git clone https://github.com/fyb
    kubectl config set-context --current --namespace=fybrik-airbyte-sample
    ```
 
-1. Create an asset (the `userdata` asset), the policy to access it (we use a policy that does not restrict access nor mandate any transformations), and an application that requires this asset:
+1. Create a policy to allow access any asset (we use a policy that does not restrict access nor mandate any transformations):
```
Comment (cdoron) [inline fybrik/README.md:36]: to allow access any asset -> to allow access to any asset

**Review by cdoron — APPROVED** [review]

**Pull request merged**